core/src/main/java/io/grpc/internal/SerializingExecutor.java

```diff
@@ -59,7 +59,7 @@ private static AtomicHelper getAtomicHelper() {
   private static final int RUNNING = -1;
 
   /** Underlying executor that all submitted Runnable objects are run on. */
-  private final Executor executor;
+  private volatile Executor executor;
```

Comment:
Note: we don't actually need this to be volatile.

Reply:
oh is it because it is only set and execute from the same network thread(eventloop)?

Reply:
It is only called from within the Runnable currently being executed (`r.run();`). That is _not_ the network thread. While a Runnable is running, `runState == RUNNING` and the network thread will not access `executor`.